Date: Sat, 18 Apr 2020 11:59:09 +0200
From: Helmut Grohne <helmut@subdivi.de>
Bug-Debian: https://bugs.debian.org/958094
Description: Do not hard code the build architecture pkg-config

--- a/src/corba/makefile
+++ b/src/corba/makefile
@@ -1,9 +1,10 @@
 
 
 CC = cc
+PKG_CONFIG ?= pkg-config
 INCFLAGS = -I../base/ -I../dynlibsrc
 #gCFLAGS = -Wall -g -pedantic -c  -DUNIX -I../base/
-#CFLAGS +=  -c  -DUNIX  -I../base/ `pkg-config --cflags glib-2.0` -I/usr/local/include/orbit-1.0/
+#CFLAGS +=  -c  -DUNIX  -I../base/ `$(PKG_CONFIG) --cflags glib-2.0` -I/usr/local/include/orbit-1.0/
 CFLAGS  += $(CPPFLAGS) -pthread -c  -DUNIX  -I../base/ -I../dynlibsrc `orbit-config --cflags client`
 LDFLAGS += -L../base/ -lm -lpthread
 AR_OPTIONS = ru
@@ -14,18 +15,18 @@ hspscan_corba.c : hspscan_corba.idl
 	orbit-idl hspscan_corba.idl
 
 test_server : test_server.o hspscan_server_impl.o hspscan_corba-common.o hspscan_corba-skels.o hspscan_corba-stubs.o corba_singleton.o 
-	cc -g -o test_server  test_server.o hspscan_server_impl.o  hspscan_corba-common.o hspscan_corba-skels.o hspscan_corba-stubs.o  corba_singleton.o ../dynlibsrc/libdyna.a ../base/libwisebase.a ../dynlibsrc/hsp.o ../dynlibsrc/subseqhash.o ../dynlibsrc/linkedlist_lookpos.o   ../dynlibsrc/libdyna_glib.a `pkg-config --libs glib-2.0` -lm -lORBit -lIIOP -lORBitutil -lpthread
+	cc -g -o test_server  test_server.o hspscan_server_impl.o  hspscan_corba-common.o hspscan_corba-skels.o hspscan_corba-stubs.o  corba_singleton.o ../dynlibsrc/libdyna.a ../base/libwisebase.a ../dynlibsrc/hsp.o ../dynlibsrc/subseqhash.o ../dynlibsrc/linkedlist_lookpos.o   ../dynlibsrc/libdyna_glib.a `$(PKG_CONFIG) --libs glib-2.0` -lm -lORBit -lIIOP -lORBitutil -lpthread
 
 scanwise_protein_index : scanwise_protein_index.o hspscan_server_impl.o hspscan_corba-common.o hspscan_corba-skels.o hspscan_corba-stubs.o corba_singleton.o 
-	cc -o scanwise_protein_index  scanwise_protein_index.o hspscan_server_impl.o  hspscan_corba-common.o hspscan_corba-skels.o hspscan_corba-stubs.o  corba_singleton.o  ../dynlibsrc/libdyna_glib.a ../dynlibsrc/libdyna.a ../base/libwisebase.a `pkg-config --libs glib-2.0` -lm -lORBit -lIIOP -lORBitutil -lpthread
+	cc -o scanwise_protein_index  scanwise_protein_index.o hspscan_server_impl.o  hspscan_corba-common.o hspscan_corba-skels.o hspscan_corba-stubs.o  corba_singleton.o  ../dynlibsrc/libdyna_glib.a ../dynlibsrc/libdyna.a ../base/libwisebase.a `$(PKG_CONFIG) --libs glib-2.0` -lm -lORBit -lIIOP -lORBitutil -lpthread
 
 
 test_client : test_client.o hspscan_corba-common.o hspscan_corba-stubs.o 
-	cc -g -o test_client  test_client.o  hspscan_corba-common.o hspscan_corba-stubs.o ../dynlibsrc/libdyna.a ../base/libwisebase.a `pkg-config --libs glib-2.0` -lm -lORBit -lIIOP -lORBitutil -lpthread
+	cc -g -o test_client  test_client.o  hspscan_corba-common.o hspscan_corba-stubs.o ../dynlibsrc/libdyna.a ../base/libwisebase.a `$(PKG_CONFIG) --libs glib-2.0` -lm -lORBit -lIIOP -lORBitutil -lpthread
 
 
 test_wrapper : test_wrapper.o hspscan_corba-common.o hspscan_corba-stubs.o hspscan_corba_wrapper.o corba_singleton.o 
-	cc -g -o test_wrapper test_wrapper.o  hspscan_corba-common.o hspscan_corba-stubs.o   corba_singleton.o hspscan_corba_wrapper.o ../dynlibsrc/hsp.o ../dynlibsrc/libdyna.a ../base/libwisebase.a  `pkg-config --libs glib-2.0` -lm -lORBit -lIIOP -lORBitutil -lpthread
+	cc -g -o test_wrapper test_wrapper.o  hspscan_corba-common.o hspscan_corba-stubs.o   corba_singleton.o hspscan_corba_wrapper.o ../dynlibsrc/hsp.o ../dynlibsrc/libdyna.a ../base/libwisebase.a  `$(PKG_CONFIG) --libs glib-2.0` -lm -lORBit -lIIOP -lORBitutil -lpthread
 
 .c.o :
 	$(CC) $(CFLAGS) $(INCFLAGS) $?
--- a/src/dnaindex/assembly/makefile
+++ b/src/dnaindex/assembly/makefile
@@ -26,10 +26,11 @@ arraymatcher : arraymatcher.o assemblygr
 
 
 CC = cc
-CFLAGS += $(CPPFLAGS) -Wall -pthread -c  -DHAS_PTHREAD_SETSCOPE -g -DUNIX -I../../base/ -I../../dynlibsrc/ -I../ -I../../models/ `pkg-config --cflags glib-2.0`
+PKG_CONFIG ?= pkg-config
+CFLAGS += $(CPPFLAGS) -Wall -pthread -c  -DHAS_PTHREAD_SETSCOPE -g -DUNIX -I../../base/ -I../../dynlibsrc/ -I../ -I../../models/ `$(PKG_CONFIG) --cflags glib-2.0`
 
 
-LDFLAGS +=  -g -L../../base/ -L../../dynlibsrc/ -lpthread `pkg-config --libs glib-2.0` -lm
+LDFLAGS +=  -g -L../../base/ -L../../dynlibsrc/ -lpthread `$(PKG_CONFIG) --libs glib-2.0` -lm
 
 
 .c.o :
--- a/src/dnaindex/makefile
+++ b/src/dnaindex/makefile
@@ -27,14 +27,14 @@ all: $(ASSEMBLY_OBJ)
 
 
 kwise : kwise.o $(ASSEMBLY_OBJ)
-	$(CC) -o kwise kwise.o $(ASSEMBLY_OBJ)  -ldyna -lwisebase $(LDFLAGS) `pkg-config --libs glib-2.0`
+	$(CC) -o kwise kwise.o $(ASSEMBLY_OBJ)  -ldyna -lwisebase $(LDFLAGS) `$(PKG_CONFIG) --libs glib-2.0`
 
 arraymatcher : arraymatcher.o $(ASSEMBLY_OBJ) 
-	$(CC) -o arraymatcher arraymatcher.o $(ASSEMBLY_OBJ)  -ldyna -lwisebase $(LDFLAGS) `pkg-config --libs glib-2.0`
+	$(CC) -o arraymatcher arraymatcher.o $(ASSEMBLY_OBJ)  -ldyna -lwisebase $(LDFLAGS) `$(PKG_CONFIG) --libs glib-2.0`
 
 
 findbad_kmer : findbad_kmer.o $(ASSEMBLY_OBJ)
-	$(CC) -o findbad_kmer findbad_kmer.o $(ASSEMBLY_OBJ)  -ldyna -lwisebase $(LDFLAGS) `pkg-config --libs glib-2.0`
+	$(CC) -o findbad_kmer findbad_kmer.o $(ASSEMBLY_OBJ)  -ldyna -lwisebase $(LDFLAGS) `$(PKG_CONFIG) --libs glib-2.0`
 
 make_shotgun : make_shotgun.o shotgun.o
 	$(CC) -o make_shotgun make_shotgun.o shotgun.o -ldyna -lwisebase $(LDFLAGS)
@@ -70,10 +70,11 @@ show_euler : show_euler.o eulerindex.o s
 	$(CC) -o show_euler show_euler.o eulerindex.o singleseqspace.o -ldyna -lwisebase $(LDFLAGS)
 
 CC = cc
-CFLAGS += $(CPPFLAGS) -pthread -c  -DHAS_PTHREAD_SETSCOPE -g -DUNIX -I../base/ -I../dynlibsrc/ `pkg-config --cflags glib-2.0`
+PKG_CONFIG ?= pkg-config
+CFLAGS += $(CPPFLAGS) -pthread -c  -DHAS_PTHREAD_SETSCOPE -g -DUNIX -I../base/ -I../dynlibsrc/ `$(PKG_CONFIG) --cflags glib-2.0`
 
 
-LDFLAGS +=  -g -L../base/ -L../dynlibsrc/ -lpthread `pkg-config --libs glib-2.0` -lm
+LDFLAGS +=  -g -L../base/ -L../dynlibsrc/ -lpthread `$(PKG_CONFIG) --libs glib-2.0` -lm
 
 
 .c.o :
--- a/src/dynlibsrc/makefile
+++ b/src/dynlibsrc/makefile
@@ -82,11 +82,12 @@ DYNHELPEROBJ = packaln.o \
 
 
 CC = cc
+PKG_CONFIG ?= pkg-config
 INCFLAGS = -I../base/
 #CFLAGS = -Wall -g -pedantic -c  -DUNIX -I../base/
-CFLAGS += $(CPPFLAGS) -Wall -pthread -g3 -DCOMPILE_VERBOSITY -DPTHREAD -c  -DUNIX -I../base/ `pkg-config --cflags glib-2.0`
-#CFLAGS +=  -Wall -g3 -pg -DPTHREAD  -c  -DUNIX -I../base/ `pkg-config --cflags glib-2.0`
-#CFLAGS +=   -c -pthread  -DUNIX  -I../base/ `pkg-config --cflags glib-2.0`
+CFLAGS += $(CPPFLAGS) -Wall -pthread -g3 -DCOMPILE_VERBOSITY -DPTHREAD -c  -DUNIX -I../base/ `$(PKG_CONFIG) --cflags glib-2.0`
+#CFLAGS +=  -Wall -g3 -pg -DPTHREAD  -c  -DUNIX -I../base/ `$(PKG_CONFIG) --cflags glib-2.0`
+#CFLAGS +=   -c -pthread  -DUNIX  -I../base/ `$(PKG_CONFIG) --cflags glib-2.0`
 #CFLAGS =   -O -c -DUNIX -DNOERROR -I../base/
 LDFLAGS +=  -g -L../base/ -lm 
 AR_OPTIONS = ru
@@ -148,13 +149,13 @@ test_shadowindex : test_shadowindex.o sh
 	$(CC) $(LDFLAGS) -g -o test_shadowindex test_shadowindex.o shadowseqindex.o shadowseq.o ./libdyna.a ../base/libwisebase.a -lm
 
 test_seqhash : test_seqhash.o subseqhash.o hsp.o hitlist.o hspscaninterface.o hsplookupscan.o libdyna.a
-	$(CC) $(LDFLAGS) -g -o test_seqhash `pkg-config --libs glib-2.0` test_seqhash.o subseqhash.o hsp.o  hitlist.o hspscaninterface.o hsplookupscan.o libdyna.a ../base/libwisebase.a -lm  
+	$(CC) $(LDFLAGS) -g -o test_seqhash `$(PKG_CONFIG) --libs glib-2.0` test_seqhash.o subseqhash.o hsp.o  hitlist.o hspscaninterface.o hsplookupscan.o libdyna.a ../base/libwisebase.a -lm  
 
 test_hsp : test_hsp.o codon.o sequence.o hsp.o compmat.o probability.o 
-	$(CC) $(LDFLAGS) -g -o test_hsp test_hsp.o codon.o sequence.o hsp.o compmat.o probability.o  ../base/libwisebase.a -lm `pkg-config --libs glib-2.0`
+	$(CC) $(LDFLAGS) -g -o test_hsp test_hsp.o codon.o sequence.o hsp.o compmat.o probability.o  ../base/libwisebase.a -lm `$(PKG_CONFIG) --libs glib-2.0`
 
 test_topscore : test_topscore.o codon.o sequence.o hsp.o compmat.o probability.o 
-	$(CC) $(LDFLAGS) -g -o test_topscore test_topscore.o codon.o sequence.o hsp.o compmat.o probability.o  ../base/libwisebase.a -lm `pkg-config --libs glib-2.0`
+	$(CC) $(LDFLAGS) -g -o test_topscore test_topscore.o codon.o sequence.o hsp.o compmat.o probability.o  ../base/libwisebase.a -lm `$(PKG_CONFIG) --libs glib-2.0`
 
 
 testgendb : probability.o genomicdb.o genomic.o testgendb.o sequence.o codon.o sequencedb.o complexsequence.o complexevalset.o
--- a/src/makefile
+++ b/src/makefile
@@ -23,6 +23,7 @@
 # Most likely you have a non ANSI C compiler
 # CC =  /opt/intel_cc_80/bin/icc
 CC = cc
+PKG_CONFIG ?= pkg-config
 
 # These are the CFLAGS to use. These days Wise2 is inherently
 # pthreaded, and everything is compiled with threads. However
@@ -31,13 +32,13 @@ CC = cc
 
 
 # Intel icc flags
-# CFLAGS += -c  -axWK -ipo `pkg-config --cflags glib-2.0`
+# CFLAGS += -c  -axWK -ipo `$(PKG_CONFIG) --cflags glib-2.0`
 
 # alpha flags (-pthread)
-# CFLAGS += -c -pthread `pkg-config --cflags glib-2.0`
+# CFLAGS += -c -pthread `$(PKG_CONFIG) --cflags glib-2.0`
 
 # normal linux/bsd/mac flags
-CFLAGS += $(CPPFLAGS) -c  `pkg-config --cflags glib-2.0`
+CFLAGS += $(CPPFLAGS) -c  `$(PKG_CONFIG) --cflags glib-2.0`
 
 
 EXTRALIBS = -lm
--- a/src/models/makefile
+++ b/src/models/makefile
@@ -192,10 +192,10 @@ psw : psw.o seqaligndisplay.o proteinsw.
 	$(CC) -o psw psw.o sw_wrap.o seqaligndisplay.o proteinsw.o abc.o pba.o   $(LDFLAGS) -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS)
 
 scanwisep.o : scanwisep.c
-	$(CC) $(CFLAGS) $(INCFLAGS) -I../external/mott `pkg-config --cflags glib-2.0` $?
+	$(CC) $(CFLAGS) $(INCFLAGS) -I../external/mott `$(PKG_CONFIG) --cflags glib-2.0` $?
 
 hsp2aln_sw.o : hsp2aln_sw.c
-	$(CC) $(CFLAGS) $(INCFLAGS) `pkg-config --cflags glib-2.0` $?
+	$(CC) $(CFLAGS) $(INCFLAGS) `$(PKG_CONFIG) --cflags glib-2.0` $?
 
 scanwisep_mysql.o : scanwisep.c
 	$(CC)  $(CFLAGS) -DSCAN_MYSQL -I../corba -I../external/mott -I../mysql/protein_index/ -I/usr/local/mysql/include scanwisep.c 
@@ -226,25 +226,25 @@ scanwisep_compress : scanwisep_compress.
 
 
 scanwisep : scanwisep.o seqaligndisplay.o proteinsw.o sw_wrap.o abc.o pba.o hsp2aln_sw.o
-	$(CC) -o scanwisep scanwisep.o sw_wrap.o seqaligndisplay.o proteinsw.o abc.o pba.o hsp2aln_sw.o   $(LDFLAGS)  -L../external/mott -lmott -ldyna_glib -ldyna  -lwisebase  $(EXTRALIBS) `pkg-config --libs glib-2.0` -lpthread
+	$(CC) -o scanwisep scanwisep.o sw_wrap.o seqaligndisplay.o proteinsw.o abc.o pba.o hsp2aln_sw.o   $(LDFLAGS)  -L../external/mott -lmott -ldyna_glib -ldyna  -lwisebase  $(EXTRALIBS) `$(PKG_CONFIG) --libs glib-2.0` -lpthread
 
 test_gwhsp : test_gwhsp.o genewisehsp.o
-	$(CC) -o test_gwhsp test_gwhsp.o genewisehsp.o $(LDFLAGS) -ldyna_glib -ldyna_glib -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS) `pkg-config --libs glib-2.0` -lpthread
+	$(CC) -o test_gwhsp test_gwhsp.o genewisehsp.o $(LDFLAGS) -ldyna_glib -ldyna_glib -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS) `$(PKG_CONFIG) --libs glib-2.0` -lpthread
 
 localcishit.o : localcishit.c
-	$(CC) $(CFLAGS) $(INCFLAGS) `pkg-config --cflags glib-2.0` $?
+	$(CC) $(CFLAGS) $(INCFLAGS) `$(PKG_CONFIG) --cflags glib-2.0` $?
 
 promoterwise.o : promoterwise.c  
-	$(CC) $(CFLAGS) $(INCFLAGS) `pkg-config --cflags glib-2.0` $?
+	$(CC) $(CFLAGS) $(INCFLAGS) `$(PKG_CONFIG) --cflags glib-2.0` $?
 
 promoterwise : promoterwise.o localdba.o localcishit.o localcispara.o dbadisplay.o motifmatrix.o motifmatrixdp.o transfactor.o pwmdna.o pairwiseshortdna.o
-	$(CC) -g -o promoterwise promoterwise.o localdba.o  localcishit.o localcispara.o dbadisplay.o  motifmatrix.o motifmatrixdp.o transfactor.o pwmdna.o  pairwiseshortdna.o $(LDFLAGS) -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS) `pkg-config --libs glib-2.0` -lpthread
+	$(CC) -g -o promoterwise promoterwise.o localdba.o  localcishit.o localcispara.o dbadisplay.o  motifmatrix.o motifmatrixdp.o transfactor.o pwmdna.o  pairwiseshortdna.o $(LDFLAGS) -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS) `$(PKG_CONFIG) --libs glib-2.0` -lpthread
 
 test_deng : test_deng.o localdba.o localcishit.o localcispara.o dbadisplay.o motifmatrix.o motifmatrixdp.o transfactor.o pwmdna.o pairwiseshortdna.o dnaprofile.o dnaprofiledp.o dnaprofileengine.o
-	$(CC) -o test_deng test_deng.o  localdba.o localcishit.o localcispara.o dbadisplay.o motifmatrix.o motifmatrixdp.o transfactor.o pwmdna.o pairwiseshortdna.o dnaprofile.o dnaprofiledp.o dnaprofileengine.o $(LDFLAGS) -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS) `pkg-config --libs glib-2.0` -lpthread
+	$(CC) -o test_deng test_deng.o  localdba.o localcishit.o localcispara.o dbadisplay.o motifmatrix.o motifmatrixdp.o transfactor.o pwmdna.o pairwiseshortdna.o dnaprofile.o dnaprofiledp.o dnaprofileengine.o $(LDFLAGS) -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS) `$(PKG_CONFIG) --libs glib-2.0` -lpthread
 
 promotercluster : promotercluster.o localdba.o localcishit.o localcispara.o dbadisplay.o motifmatrix.o motifmatrixdp.o transfactor.o pwmdna.o pairwiseshortdna.o dnaprofile.o dnaprofiledp.o dnaprofileengine.o
-	$(CC) -o promotercluster promotercluster.o  localdba.o localcishit.o localcispara.o dbadisplay.o motifmatrix.o motifmatrixdp.o transfactor.o pwmdna.o pairwiseshortdna.o dnaprofile.o dnaprofiledp.o dnaprofileengine.o $(LDFLAGS) -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS) `pkg-config --libs glib-2.0` -lpthread
+	$(CC) -o promotercluster promotercluster.o  localdba.o localcishit.o localcispara.o dbadisplay.o motifmatrix.o motifmatrixdp.o transfactor.o pwmdna.o pairwiseshortdna.o dnaprofile.o dnaprofiledp.o dnaprofileengine.o $(LDFLAGS) -ldyna_glib -ldyna -lwisebase  $(EXTRALIBS) `$(PKG_CONFIG) --libs glib-2.0` -lpthread
 
 dnawise : dnawise.o dnahmm.o dnahmmdp.o seqaligndisplay.o 
 	$(CC) -o dnawise dnawise.o dnahmm.o dnahmmdp.o  seqaligndisplay.o $(LDFLAGS) -ldyna_glib -ldyna -lwisebase  -lpthread
@@ -343,13 +343,14 @@ HMMER_DEFINE =
 
 
 CC = cc
+PKG_CONFIG ?= pkg-config
 #CFLAGS = -c -O2  -pg -I../base/ -I../dynlibsrc/
 #CFLAGS = -c -O2 -DPTHREAD -DHAS_PTHREAD_SETSCOPE  -DUNIX  -I../base/ -I../dynlibsrc/ -I.
-CFLAGS += -c -g3  -DUNIX  -I../base/ -I../dynlibsrc/ -I. -pthread `pkg-config --cflags glib-2.0`
-#CFLAGS += -c  -g -DHAS_PTHREAD_SETSCOPE -DUNIX -I../base/ -I../dynlibsrc/ `pkg-config --cflags glib-2.0`
+CFLAGS += -c -g3  -DUNIX  -I../base/ -I../dynlibsrc/ -I. -pthread `$(PKG_CONFIG) --cflags glib-2.0`
+#CFLAGS += -c  -g -DHAS_PTHREAD_SETSCOPE -DUNIX -I../base/ -I../dynlibsrc/ `$(PKG_CONFIG) --cflags glib-2.0`
 #LDFLAGS =  -L../base/ -L../dynlibsrc/ -L../HMMer2/  
 
-LDFLAGS +=  -L../base/ -L../dynlibsrc/ -L$(HMMER_LIBS) -lpthread `pkg-config --libs glib-2.0`
+LDFLAGS +=  -L../base/ -L../dynlibsrc/ -L$(HMMER_LIBS) -lpthread `$(PKG_CONFIG) --libs glib-2.0`
 AR_OPTIONS = ru
 
 RANLIB_NEEDED = 0
--- a/src/network/makefile
+++ b/src/network/makefile
@@ -2,9 +2,10 @@
 .SUFFIXES :  .dy
 
 CC = cc
+PKG_CONFIG ?= pkg-config
 INCFLAGS = -I../base/  -I../socket -I../dynlibsrc -I../dnaindex
-CFLAGS += $(CPPFLAGS)  -c -pthread  -DUNIX  -I../base/ -I../socket -I../dynlibsrc -I../dnaindex `pkg-config --cflags glib-2.0`
-LDFLAGS +=  -g -L../base/ -L../socket -L../dynlibsrc -L../dnaindex -lm `pkg-config --libs glib-2.0`
+CFLAGS += $(CPPFLAGS)  -c -pthread  -DUNIX  -I../base/ -I../socket -I../dynlibsrc -I../dnaindex `$(PKG_CONFIG) --cflags glib-2.0`
+LDFLAGS +=  -g -L../base/ -L../socket -L../dynlibsrc -L../dnaindex -lm `$(PKG_CONFIG) --libs glib-2.0`
 AR_OPTIONS = ru
 
 RANLIB_NEEDED = 0
--- a/src/other_programs/makefile
+++ b/src/other_programs/makefile
@@ -3,8 +3,9 @@
 .SUFFIXES :  .dy
 
 CC = cc
-CFLAGS += $(CPPFLAGS) -g3 -c -pthread  -DUNIX  -I../base/ -I../dynlibsrc `pkg-config --cflags glib-2.0`
-LDFLAGS += -g -L../base/ -L../dynlibsrc  -lm `pkg-config --libs glib-2.0`
+PKG_CONFIG ?= pkg-config
+CFLAGS += $(CPPFLAGS) -g3 -c -pthread  -DUNIX  -I../base/ -I../dynlibsrc `$(PKG_CONFIG) --cflags glib-2.0`
+LDFLAGS += -g -L../base/ -L../dynlibsrc  -lm `$(PKG_CONFIG) --libs glib-2.0`
 AR_OPTIONS = ru
 
 RANLIB_NEEDED = 0
--- a/src/snp/makefile
+++ b/src/snp/makefile
@@ -3,11 +3,12 @@
 
 
 CC = cc
+PKG_CONFIG ?= pkg-config
 #CFLAGS = -c -O2  -pg -I../base/ -I../dynlibsrc/
 #CFLAGS = -c -O2 -DPTHREAD -DHAS_PTHREAD_SETSCOPE  -DUNIX  -I../base/ -I../dynlibsrc/ -I.
-CFLAGS += $(CPPFLAGS) -c -g3  -DUNIX  -I../base/ -I../dynlibsrc/ -I. -pthread `pkg-config --cflags glib-2.0`
+CFLAGS += $(CPPFLAGS) -c -g3  -DUNIX  -I../base/ -I../dynlibsrc/ -I. -pthread `$(PKG_CONFIG) --cflags glib-2.0`
 
-LDFLAGS +=  -g -L../base/ -L../dynlibsrc/ -lpthread `pkg-config --libs glib-2.0` -lpthread -lm
+LDFLAGS +=  -g -L../base/ -L../dynlibsrc/ -lpthread `$(PKG_CONFIG) --libs glib-2.0` -lpthread -lm
 AR_OPTIONS = ru
 
 RANLIB_NEEDED = 0
